src/compiler/compile/Component.ts

```diff
@@ -1284,10 +1284,23 @@ export default class Component {
 				const module_dependencies = new Set<string>();
 
 				let scope = this.instance_scope;
+				const {declarations: outsetScopeDecalarations} = this.instance_scope;
 				const map = this.instance_scope_map;
 
 				walk(node.body, {
 					enter(node: Node, parent) {
+						if (node.type === 'VariableDeclaration' && node.kind === 'var') {
+							const names = extract_names(node.declarations[0].id);
+							const is_var_in_outset = names.every((i: string) => {
```

Comment:
shouldn't this be `some`? if at least 1 variable is declared in the `outsetScopeDecalarations`, you should then throw compile error